src/moonlink/src/storage/iceberg/rest_catalog.rs

```diff
@@ -0,0 +1,291 @@
+use async_trait::async_trait;
+use iceberg::table::Table;
+use iceberg::CatalogBuilder;
+use iceberg::Result as IcebergResult;
+use iceberg::{Catalog, Namespace, NamespaceIdent, TableCommit, TableCreation, TableIdent};
+use iceberg_catalog_rest::{
+    RestCatalog as IcebergRestCatalog, RestCatalogBuilder as IcebergRestCatalogBuilder,
+    REST_CATALOG_PROP_URI, REST_CATALOG_PROP_WAREHOUSE,
+};
+use reqwest::Client;
+use std::collections::HashMap;
+
+#[derive(Debug)]
+pub struct RestCatalog {
+    pub(crate) catalog: IcebergRestCatalog,
+}
+
+#[derive(Debug, Default)]
+pub struct RestCatalogConfig {
```

Comment:
Wait I think we already have one
https://github.com/Mooncake-Labs/moonlink/blob/94656f903a8fb91565b6fe41b7ebcdef6d9729e8/src/moonlink/src/storage/iceberg/iceberg_table_config.rs#L68